concrete/src/File/Image/Svg/Sanitizer.php

```diff
@@ -0,0 +1,148 @@
+<?php
+
+namespace Concrete\Core\File\Image\Svg;
+
+use DOMDocument;
+use DOMElement;
+use Exception;
+use Throwable;
+
+class Sanitizer
+{
+    /**
+     * Sanitize a file containing a SVG document.
+     *
+     * @param string $inputFilename The input filename
+     * @param \Concrete\Core\File\Image\Svg\SanitizerOptions $options the sanitizer options (if NULL, we'll use the default ones)
+     * @param string $outputFilename The output filename (if empty, we'll overwrite $inputFilename)
+     *
+     * @throws \Concrete\Core\File\Image\Svg\SanitizerException in case of errors
+     */
+    public function sanitizeFile($inputFilename, SanitizerOptions $options = null, $outputFilename = '')
+    {
+        $data = is_string($inputFilename) && is_file($inputFilename) ? @file_get_contents($inputFilename) : false;
+        if ($data === false) {
+            throw SanitizerException::create(SanitizerException::ERROR_FAILED_TO_READ_FILE);
+        }
+        $sanitizedData = $this->sanitizeData($data, $options);
+        if ((string) $outputFilename === '') {
+            $outputFilename = $inputFilename;
+        }
+        if ($outputFilename !== $inputFilename || $data !== $sanitizedData) {
+            if (@file_put_contents($outputFilename, $sanitizedData) === false) {
```

Comment:
It's probably worth using a filesystem instance here, either League or Illuminate because that gives you the ability to use test doubles to mock this functionality.

```php
$filesystem = \Mockery::mock(\Illuminate\Filesystem\Filesystem::class)->shouldRecieve('put')->withArgs($sanitized)->once();
```